xc7/utils/prjxray_create_edges.py

```diff
@@ -89,57 +70,63 @@ def add_graph_nodes_for_pins(conn, tile_type, wire, pin_directions):
     else:
         assert False, pin_direction
 
-    # Find all instances of this specific wire.
-    c.execute(
-        """
-        SELECT pkey, node_pkey, tile_pkey
-            FROM wire WHERE wire_in_tile_pkey = ?;""", (wire_in_tile_pkey, )
-    )
-
     c2 = conn.cursor()
-    c3 = conn.cursor()
     c2.execute("""BEGIN EXCLUSIVE TRANSACTION;""")
 
-    for wire_pkey, node_pkey, tile_pkey in c:
-        c3.execute(
+    for wire_in_tile_pkey in wire_in_tile_pkeys.values():
+        # Find all instances of this specific wire.
+        c.execute(
             """
-            SELECT grid_x, grid_y FROM tile WHERE pkey = ?;""", (tile_pkey, )
+            SELECT pkey, node_pkey, tile_pkey
+                FROM wire WHERE wire_in_tile_pkey = ?;""",
+            (wire_in_tile_pkey, )
         )
 
-        grid_x, grid_y = c3.fetchone()
-
-        updates = []
-        values = []
+        c3 = conn.cursor()
```

Comment:
Could you give these cursors better names? Maybe something like `read_cursor` and `write_cursor`. It wasn't immediately clear why `c2` commits at the end and `c3` doesn't.

Reply:
Good idea, I've updated the write cursors.